gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractNativeMacApplicationPackageDmgTask.kt

```diff
@@ -0,0 +1,144 @@
+ */
+
+package org.jetbrains.compose.desktop.application.tasks
+
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import org.jetbrains.compose.desktop.application.internal.ioFile
+import org.jetbrains.compose.desktop.application.internal.notNullProperty
+import java.io.File
+
+abstract class AbstractNativeMacApplicationPackageDmgTask : AbstractNativeMacApplicationPackageTask() {
+    @get:InputFile
+    @get:PathSensitive(PathSensitivity.ABSOLUTE)
+    val hdiutil: RegularFileProperty = objects.fileProperty().value { File("/usr/bin/hdiutil") }
+
+    @get:InputFile
+    @get:PathSensitive(PathSensitivity.ABSOLUTE)
+    val osascript: RegularFileProperty = objects.fileProperty().value { File("/usr/bin/osascript") }
+
+    @get:Input
+    val installDir: Property<String> = objects.notNullProperty("/Applications")
+
+    @get:InputDirectory
+    val appDir: DirectoryProperty = objects.directoryProperty()
+
+    override fun createPackage(destinationDir: File, workingDir: File) {
+        val packageName = packageName.get()
+        val fullPackageName = fullPackageName.get()
+        val tmpImage = workingDir.resolve("$fullPackageName.tmp.dmg")
+        val finalImage = destinationDir.resolve("$fullPackageName.dmg")
+
+        createImage(volumeName = packageName, imageFile = tmpImage, srcDir = appDir.ioFile)
+        val mounted = mountImage(volumeName = packageName, imageFile = tmpImage)
+        try {
+            runSetupScript(appName = packageName, mounted)
+        } finally {
+            unmountImage(mounted)
+        }
+        finalizeImage(tmpImage, finalImage)
+        logger.lifecycle("The distribution is written to ${finalImage.canonicalPath}")
+    }
+
+    private fun createImage(volumeName: String, imageFile: File, srcDir: File) {
+        var size = srcDir.walk().filter { it.isFile }.sumOf { it.length() }
+        size += 10 * 1024 * 1024
+
+        hdiutil(
+            "create",
+            "-srcfolder", srcDir.absolutePath,
+            "-volname", volumeName,
+            "-size", size.toString(),
+            "-ov", imageFile.absolutePath,
+            "-fs", "HFS+",
+            "-format", "UDRW"
+        )
+    }
+
+    private data class MountedImage(val device: String, val disk: String)
+    private fun mountImage(volumeName: String, imageFile: File): MountedImage {
+        val output = hdiutil(
+            "attach",
+            "-readwrite",
+            "-noverify",
+            "-noautoopen",
+            imageFile.absolutePath
+        )
+        Thread.sleep(3000)
+        var device: String? = null
+        val volume: String = output.substring(output.lastIndexOf("/Volumes/$volumeName")).trim()
+
+        for (entry in output.split("\\s+".toRegex())) {
+            if (entry.startsWith("/dev/")) {
+                device = entry
```

Comment:
Could there be multiple entries like that?

Reply:
Maybe, fixed